internal/cli/logs_test.go

```diff
@@ -1,16 +1,144 @@
 package cli
 
 import (
+	"bytes"
+	"fmt"
+	"io"
 	"testing"
 	"time"
 
 	"github.com/auth0/go-auth0/management"
+	"github.com/golang/mock/gomock"
 
 	"github.com/stretchr/testify/assert"
 
 	"github.com/auth0/auth0-cli/internal/auth0"
+	"github.com/auth0/auth0-cli/internal/auth0/mock"
+	"github.com/auth0/auth0-cli/internal/display"
 )
 
+func TestTailLogsCommand(t *testing.T) {
+	t.Run("it returns early with a message to generate logs when there are no logs", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		logsAPI := mock.NewMockLogAPI(ctrl)
+		logsAPI.EXPECT().
+			List(gomock.Any()).
+			Return([]*management.Log{}, nil)
+
+		expectedResult := `No logs available.
+
+ ▸    To generate logs, run a test command like 'auth0 test login' or 'auth0 test token'
+`
+
+		stdout := &bytes.Buffer{}
+		cli := &cli{
+			renderer: &display.Renderer{
+				MessageWriter: stdout,
+				ResultWriter:  io.Discard,
+			},
+			api: &auth0.API{Log: logsAPI},
+		}
+
+		cmd := tailLogsCmd(cli)
+		cmd.SetArgs([]string{"--number", "9000", "--filter", "user_id:123"})
+		err := cmd.Execute()
+
+		assert.NoError(t, err)
+		assert.Equal(t, expectedResult, stdout.String())
+	})
+
+	t.Run("it returns an error when it fails to get the logs on the first request", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		logsAPI := mock.NewMockLogAPI(ctrl)
+		logsAPI.EXPECT().
+			List(gomock.Any()).
+			Return(nil, fmt.Errorf("generic error"))
+
+		cli := &cli{
+			api: &auth0.API{Log: logsAPI},
+		}
+
+		cmd := tailLogsCmd(cli)
+		cmd.SetArgs([]string{"--number", "9000", "--filter", "user_id:123"})
+		err := cmd.Execute()
+
+		assert.EqualError(t, err, "failed to get logs: generic error")
+	})
+
+	t.Run("it returns an error when it fails to get the logs on the 3rd request", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		logsAPI := mock.NewMockLogAPI(ctrl)
+		logsAPI.EXPECT().
+			List(gomock.Any()).
+			Return(
+				[]*management.Log{
+					{
+						LogID:       auth0.String("354234"),
+						Type:        auth0.String("sapi"),
+						Description: auth0.String("Update branding settings"),
+					},
+				},
+				nil,
+			)
+
+		logsAPI.EXPECT().
+			List(gomock.Any()).
+			Return(
+				[]*management.Log{
+					{
+						LogID:       auth0.String("354234"),
+						Type:        auth0.String("sapi"),
+						Description: auth0.String("Update branding settings"),
+					},
+					{
+						LogID:       auth0.String("354236"),
+						Type:        auth0.String("sapi"),
+						Description: auth0.String("Update tenant settings"),
+					},
+				},
+				nil,
+			)
+
+		logsAPI.EXPECT().
+			List(gomock.Any()).
+			Return(nil, fmt.Errorf("generic error"))
+
+		expectedMessage := `
+=== auth0-cli-tests.eu.auth0.com logs
+
+ ▸    Failed to get latest logs: generic error
```

Comment:
Maybe just assert that the output contains `Failed to get latest logs: generic error` to avoid breaking the test if the bullet (`▸`) or the header (`===`...) change in the future?